pkg/scheduler/factory.go

```diff
@@ -297,14 +299,17 @@ func (c *Configurator) createFromPolicy(policy schedulerapi.Policy) (*Scheduler,
 	if pluginConfig, err = dedupPluginConfigs(pluginConfig); err != nil {
 		return nil, err
 	}
-	for i := range c.profiles {
-		prof := &c.profiles[i]
-		// Plugins and PluginConfig are empty when using Policy; overriding.
-		prof.Plugins = &schedulerapi.Plugins{}
-		prof.Plugins.Append(&plugins)
-		prof.PluginConfig = pluginConfig
+	c.profiles = []schedulerapi.KubeSchedulerProfile{
+		{
+			SchedulerName: v1.DefaultSchedulerName,
+			Plugins:       &plugins,
+			PluginConfig:  pluginConfig,
+		},
 	}
 
+	// Use the latest version to set the default values for plugin configs.
+	v1beta2.ConvertToInternalPluginConfigArgs(&c.profiles[0])
```

Comment:
But these are all internal types already. This function does nothing, I think.

I see two alternatives:
- make sure that the pluginconfigs that the legacy registry returns include the proper defaults.
- Change the legacy registry to use external types and do a conversion.

Reply:
This function does the plugin config defaulting.

Reply:
So I copied the plugin defaulting logic here. It should be fine for now, in all cases policy should have its own defaulting.